pkg/autoscaler/autoscaler.go

```diff
@@ -291,6 +294,106 @@ var _ = Describe("Autoscaler should", framework.LabelAutoscaler, Serial, func()
+				asr := machineAutoscalerResource(machineSet, 0, expectedReplicas)
+				Expect(client.Create(ctx, asr)).Should(Succeed(), fmt.Sprintf("Failed to create MachineAutoscaler with min 0/max %v replicas", expectedReplicas))
+				cleanupObjects[asr.GetName()] = asr
+				arch := machineSetParams.Labels[framework.ArchLabel]
+				if arch != framework.Amd64 || workloadArch == "" {
+					// the workloadArch == "" condition ensures that workloadArch and expectedScaledMachineSet are set
+					// at least once. This is needed to ensure that the test does not fail when there is only one amd64 MachineSet
+					workloadArch = arch
+					expectedScaledMachineSet = machineSet
+				}
+			}
+			uniqueJobName := fmt.Sprintf("%s-scale-from-zero", workloadJobName)
+			By(fmt.Sprintf("Creating scale-out workload %s: jobs: %v, memory: %s", uniqueJobName,
+				expectedReplicas+1, workloadMemRequest.String()))
+			// Executing one additional replica to ensure that the workload does not triggers other scale-out events with
+			// other MachineSets not having the requested architecture.
+			workload := framework.NewWorkLoad(expectedReplicas+1, workloadMemRequest, uniqueJobName, autoscalingTestLabel, "",
+				corev1.NodeSelectorRequirement{
+					Key:      targetedNodeLabel,
+					Operator: corev1.NodeSelectorOpExists,
+				},
+				corev1.NodeSelectorRequirement{
+					Key:      "kubernetes.io/arch",
+					Operator: corev1.NodeSelectorOpIn,
+					Values:   []string{workloadArch},
+				})
+
+			cleanupObjects[workload.GetName()] = workload
+			Expect(client.Create(ctx, workload)).Should(Succeed(), "Failed to create scale-out workload %s", workloadJobName)
+			passed := framework.RunCheckUntil(ctx,
+				func(ctx context.Context, g framework.GomegaAssertions) bool { // Continuous check condition
+					condition := true
+					for _, machineSet := range machineSets {
+						if machineSet.GetName() == expectedScaledMachineSet.GetName() {
+							// Ignore the MachineSet with the architecture requested by the workload.
+							continue
+						}
+						ms, err := framework.GetMachineSet(ctx, client, machineSet.GetName())
+						Expect(err).ToNot(HaveOccurred(), "Failed to get MachineSet %s", machineSet.GetName())
+
+						By(fmt.Sprintf("Current replicas for %s are %d, expected %d.", machineSet.GetName(),
+							*ms.Spec.Replicas, 0))
+
+						condition = condition && *ms.Spec.Replicas == 0
+					}
+
+					return condition
+				}, func(ctx context.Context, g framework.GomegaAssertions) bool { // Until condition
+					ms, err := framework.GetMachineSet(ctx, client, expectedScaledMachineSet.GetName())
+					Expect(err).ToNot(HaveOccurred(), "Failed to get MachineSet %s", expectedScaledMachineSet.GetName())
+
+					By(fmt.Sprintf("Waiting for machineSet replicas to scale out. Current replicas are %v, expected %v.",
+						*ms.Spec.Replicas, expectedReplicas))
+
+					return *ms.Spec.Replicas == expectedReplicas
```

Comment:
Use a gomega assertion so that we get a nice error printed out
```suggestion
return g.Expect(ms.Spec.Replicas).To(HaveValue(int32(expectedReplicas))
```